src/email.rs

```diff
@@ -42,6 +42,7 @@ pub fn request(app: &AppConfig, email_addr: EmailAddress, client_id: &str, nonce
     // Store data for this request in Redis, to reference when user uses
     // the generated link.
     try!(app.store.store_session(&session, &[
+        ("type", "email"),
```

Comment:
I don't think that would protect against te issue I'm trying to prevent here, because the session ID is visible to the user. E.g.:

 - Start an OIDC flow, and take the session ID from the `state` param in the auth redirect URL.
 - Submit to the email flow `/confirm` route, panics trying to access `code` on the session data.